akka-sample-kafka-to-sharding-scala/processor/src/main/scala/sample/sharding/kafka/UserGrpcService.scala

```diff
@@ -1,25 +1,18 @@
 package sample.sharding.kafka
 
-import akka.actor.typed.ActorRef
-import akka.actor.typed.ActorSystem
 import akka.actor.typed.scaladsl.AskPattern._
-import akka.cluster.sharding.typed.ShardingMessageExtractor
+import akka.actor.typed.{ActorRef, ActorSystem, Scheduler}
 import akka.util.Timeout
-import com.typesafe.config.ConfigFactory
-import sample.sharding.kafka.UserEvents.GetRunningTotal
-import sample.sharding.kafka.UserEvents.RunningTotal
+import sample.sharding.kafka.UserEvents.{GetRunningTotal, Message, RunningTotal}
 
-import scala.concurrent.Future
 import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContextExecutor, Future}
 
-class UserGrpcService(system: ActorSystem[_], extractor: ShardingMessageExtractor[UserEvents.Message, UserEvents.Message]) extends UserService {
+class UserGrpcService(system: ActorSystem[_], shardRegion: ActorRef[Message]) extends UserService {
```

Comment:
So now other parts of the system only know about the shardRegion actor, rather than the message extractor